### PR TITLE
feat: support a flatter input message type

### PR DESCRIPTION
**Pull request by debajyoti-truefoundry**

### CODE REVIEW

File: async_processor/app.py [hunk 1]
```diff
@@ -105,9 +100,17 @@ def _ready_route_handler(self):
...
+    async def _process_route_handler(
+        self,
+        request: Request,
+        # We need this ` _: dict = Body(...)` to
+        # ensure that the user can put request body in the Swagger page
+        _: dict = Body(...),  # noqa: B008
+    ) -> Response:
         start = time.perf_counter()
-        output = self._processor.process(body)
+        payload = await request.body()
```
Comment (chiragjn) [inline async_processor/app.py:111]: This will return the same thing as `_: dict = Body(...)`, right ?

File: async_processor/app.py [hunk 2]
```diff
@@ -105,9 +100,17 @@ def _ready_route_handler(self):
...
+        input_message = self._processor.input_deserializer(payload)
```
Comment (chiragjn) [inline async_processor/app.py:112]: Does `request.body()` also do some decoding and `input_deserializer` expects bytes?

File: async_processor/processor.py [hunk 3]
```diff
@@ -23,8 +30,18 @@ async def _async_run(
 class BaseProcessor:
     def input_deserializer(
         self, serialized_input_message: Union[str, bytes]
-    ) -> InputMessage:
-        return InputMessage(**orjson.loads(serialized_input_message))
+    ) -> InputMessageBase:
+        input_message = orjson.loads(serialized_input_message)
+        if not isinstance(input_message, dict):
+            raise ValueError(
+                'Input message must be an object (Ex: `{{"a": "b"}}`). '
```
Comment (chiragjn) [inline async_processor/processor.py:37]:
```suggestion
                'Input message must be an object (E.g.: `{"a": "b"}`). '
```

File: async_processor/app.py (hunk 1 above)
Comment (debajyoti-truefoundry) [reply async_processor/app.py:111]: This will be in binary.

File: async_processor/app.py (hunk 2 above)
Comment (debajyoti-truefoundry) [reply async_processor/app.py:112]: `input_deserializer` can work with both strings and bytes.

File: async_processor/types.py [hunk 4]
```diff
@@ -12,17 +12,69 @@ class ProcessStatus(str, enum.Enum):
...
+        return body
+
 
 class OutputMessage(BaseModel):
-    request_id: str
+    request_id: Optional[str]
     status: ProcessStatus
     body: Optional[Any] = None
     error: Optional[str] = None
+    input_message: Optional[Union[InputMessage, InputMessageV2]] = None
```
Comment (chiragjn) [inline async_processor/types.py:77]: This will only be available in failure cases?

**Review by chiragjn — APPROVED** [review]

File: async_processor/types.py [hunk 5]
```diff
@@ -12,17 +12,69 @@ class ProcessStatus(str, enum.Enum):
...
+
+    def get_body(self) -> Any:
+        return self.body
+
+
+# We cannot maintain two different types and should remove `InputMessage`
+# after sometime
+class InputMessageV2(InputMessageBase):
+    tfy_request_id: Optional[constr(regex=r"^[a-zA-Z0-9\-]{1,36}$")] = None
+    tfy_published_at_epoch_ns: Optional[int] = None
```
Comment (chiragjn) [inline async_processor/types.py:51]: Just for my understanding,
In this case the service enqueuing is responsible for setting these in the payload right?
- Without tfy_request_id they can't specifically subscribe/look for corresponding output message
- Without tfy_published_at_epoch_ns they lose out on queue wait time telemetry
Antyhing else?

File: async_processor/types.py (hunk 4 above)
Comment (debajyoti-truefoundry) [reply async_processor/types.py:77]: Yes.

File: async_processor/types.py (hunk 5 above)
Comment (debajyoti-truefoundry) [reply async_processor/types.py:51]: > In this case the service enqueuing is responsible for setting these in the payload right?

Yes

Comment (debajyoti-truefoundry) [reply async_processor/types.py:51]: > Without tfy_request_id they can't specifically subscribe/look for corresponding output message

Yes. If they are using NATS as output.

Comment (debajyoti-truefoundry) [reply async_processor/types.py:51]: > Without tfy_published_at_epoch_ns they lose out on queue wait time telemetry

Yes.

**Review by nikp1172 — APPROVED** [review]

**Pull request merged**